web/client/components/app/__tests__/StandardApp-test.jsx

```diff
@@ -53,16 +53,14 @@ describe('StandardApp', () => {
         expect(app).toExist();
     });
 
-    it('creates a default app with onInit', (done) => {
+    it('creates a default app with onInit', () => {
         const init = {
-            onInit: () => { }
+            onInit: (cfg) => {
+                expect(cfg).toExist();
+            }
         };
-        const spy = expect.spyOn(init, 'onInit');
-        ReactDOM.render(<StandardApp onInit={init.onInit}/>, document.getElementById("container"));
-        setTimeout(() => {
-            expect(spy).toHaveBeenCalled();
-            done();
-        }, 100);
+        let app = ReactDOM.render(<StandardApp onInit={init.onInit}/>, document.getElementById("container"));
```

Comment:
I think this changes the original test: you are removing the spy test of onInit. If onInit is not called this test is successful anyway.

Reply:
you're right. but there is a way to avoid to use the setTimeout?

Reply:
i was not able to find a final and good solution. Probably we can increase the timeout but that is not the solution

Reply:
did you try to remove the spy AND move done() inside the onInit function ?

Reply:
yes, but as you said before: "If onInit is not called this test is successful anyway."  the done() will not be called either

Reply:
But then it will fail for the right reason: bacause onInit is not called. If onInit is called the test will pass, whatever the time needed (within the 2 seconds max timeout of every test)

Reply:
Dammit I was so close!  

You are superb!